google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreOptions.java

```diff
@@ -208,6 +216,32 @@ public Builder setTimestampsInSnapshotsEnabled(boolean value) {
     @Override
     @Nonnull
     public FirestoreOptions build() {
+      if (System.getProperty(FIRESTORE_EMULATOR_SYSTEM_PROPERTY) != null) {
+        String emulatorHost = System.getProperty(FIRESTORE_EMULATOR_SYSTEM_PROPERTY);
+        String hostUrlString = "http://" + emulatorHost;
+        // Try creating a host in order to validate that the host name is valid.
+        try {
+          new URL(hostUrlString);
```

Comment:
Is the intention here to force a DNS probe, or just to validate that the string given to us is well formatted? If we aren't trying to force a DNS probe, please use `URI.create` here instead of `new URL`.